src/models/CredentialEngine.ts

```diff
@@ -104,6 +104,27 @@ export class CredentialEngine {
 		}
 	}
 
+	/**
+	 * Create a new DID with user metamask address as controller
+	 * @param controller
+	 * @returns {Promise<{didDocument: object, keyPair: object}>} The created DID document and key pair.
+	 * @throws Will throw an error if DID creation fails.
+	 */
+	public async createWalletDID(controller: string): Promise<{ didDocument: DidDocument; keyPair: KeyPair }> {
```

Comment:
i'm not familiar with the keypair terms but maybe 'address' or 'walletAddress' would make a more readable argument name.